backend/app/database/supabase/client.py

```
@@ -1,13 +1,18 @@
-)
+_client: AsyncClient | None = None
 
 def get_supabase_client() -> AsyncClient:
+    global _client
     """
     Returns a shared asynchronous Supabase client instance.
     """
-    return supabase_client
+    if _client is None:
+        if not settings.supabase_url or not settings.supabase_key:
+            raise RuntimeError("Supabase is not configured")
```

Ruff (0.14.14)

Avoid specifying long messages outside the exception class

(TRY003)